py2puml/inspection/inspectclass.py

```diff
@@ -83,7 +74,37 @@ def inspect_static_attributes(
             uml_attr = UmlAttribute(attr_name, attr_type, static=True)
             definition_attrs.append(uml_attr)
 
-    return definition_attrs
+
+def inspect_methods(
+    definition_methods, class_type,
+):
+    no_dunder = lambda x: not (x[0].startswith('__') or x[0].endswith('__'))
+    methods = filter(no_dunder, inspect.getmembers(class_type, callable))
+    for name, method in methods:
+        signature = inspect.signature(method)
+        uml_method = UmlMethod(
+            name=name,
+            signature=str(signature),
```

Comment:
it is a convenient way to get the whole signature, but:
- it does not output the signature in the PlantUML expected syntax see comments `2.` in https://github.com/lucsorel/py2puml/issues/11#issuecomment-1104471337
- the way `str(signature)` behave is likely to change and break the feature

However, I don't like how I implemented the the way attributes detected by inspection (dataclasses, class attributes) are typed: `builtins.date` would be displayed instead of just `date` for example. I did a better job when detecting instance attributes in constructors, the whole would need harmonization.

Reply:
> it is a convenient way to get the whole signature, but:
> 
>     * it does not output the signature in the PlantUML expected syntax see comments `2.` in [extract methods #11 (comment)](https://github.com/lucsorel/py2puml/issues/11#issuecomment-1104471337)

it does not guarantee it for sure. I should just iterate the structure we can rely on and build it from ther

> 
>     * the way `str(signature)` behave is likely to change and break the feature

It might. yeah
> 
> 
> However, I don't like how I implemented the the way attributes detected by inspection (dataclasses, class attributes) are typed: `builtins.date` would be displayed instead of just `date` for example. I did a better job when detecting instance attributes in constructors, the whole would need harmonization.

I'll make sure to take a closer look there.

Reply:
for now, let's leave the signature format like that, it is not how UML users expect it, but it is readable to python users